soil_api/models/soil_property.py

```diff
@@ -0,0 +1,233 @@
+from __future__ import annotations
+
+from enum import Enum
+from typing import List
+
+from pydantic import BaseModel, Field
+
+from soil_api.models.shared import FeatureType, PointGeometry
+
+
+class SoilPropertiesNames(Enum):
```

Comment:
Consider if you could make these into a dict as well. Seems to me like the user only needs to enter a SoilPropertiesCodes?